src/main/kotlin/no/nav/tiltakspenger/saksbehandling/behandling/service/behandling/brev/ForhåndsvisVedtaksbrevService.kt

```diff
@@ -51,23 +53,46 @@ class ForhåndsvisVedtaksbrevService(
 
         return when (behandling.behandlingstype) {
             Behandlingstype.FØRSTEGANGSBEHANDLING -> {
-                genererInnvilgelsesbrevClient.genererInnvilgelsesvedtaksbrevMedTilleggstekst(
-                    hentBrukersNavn = personService::hentNavn,
-                    hentSaksbehandlersNavn = navIdentClient::hentNavnForNavIdent,
-                    vedtaksdato = LocalDate.now(),
-                    tilleggstekst = kommando.fritekstTilVedtaksbrev,
-                    fnr = sak.fnr,
-                    saksbehandlerNavIdent = behandling.saksbehandler,
-                    beslutterNavIdent = behandling.beslutter,
-                    innvilgelsesperiode = virkingsperiode!!,
-                    saksnummer = sak.saksnummer,
-                    sakId = sak.id,
-                    forhåndsvisning = true,
-                    barnetilleggsPerioder = kommando.barnetillegg?.let { if (it.isEmpty()) null else it },
-                ).fold(
-                    ifLeft = { throw IllegalStateException("Kunne ikke generere vedtaksbrev. Underliggende feil: $it") },
-                    ifRight = { it.pdf },
-                )
+                when (kommando.utfall) {
+                    Behandlingsutfall.INNVILGELSE -> genererInnvilgelsesbrevClient.genererInnvilgelsesvedtaksbrevMedTilleggstekst(
```

Comment:
Trekke ut disse to i private function's?

Reply:
Eller, det spørs hvor stor parameter lista blir. Kanskje ikke funksjonen kymper så mye uansett.